arm-mediaservices/2015-10-01/swagger/media.json

```diff
@@ -0,0 +1,679 @@
+            "maxLength": 24
+          },
+          {
+            "name": "SyncStorageKeysInput",
+            "description": "Properties needed to sycnronize the keys for a storage account to the Media Service.",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SyncStorageKeysInput"
+            }
+          }
+        ]
+      }
+    }
+  },
+  "definitions": {
+    "ApiEndpoint": {
+      "description": "The properties for a Media Services REST API endpoint.",
+      "properties": {
+        "endpoint": {
+          "description": "The Media Services REST endpoint.",
+          "type": "string"
+        },
+        "majorVersion": {
+          "description": "The version of Media Services REST API.",
+          "type": "string"
+        }
+      }
+    },
+    "ApiError": {
+    "description": "The error returned from a failed Media Services REST API call.",
+      "properties": {
+        "code": {
+          "description": "Error code.",
+          "type": "string"
+        },
+        "message": {
+          "description": "Error message",
+          "type": "string"
+        }
+      }
+    },
+    "MediaService": {
+      "description": "The properties of a Media Service resource.",
+      "properties": {
+        "properties": {
+          "$ref": "#/definitions/MediaServiceProperties"
+        }
+      },
+      "allOf": [
+        {
+          "$ref": "#/definitions/TrackedResource"
+        }
+      ]
+    },
+    "MediaServiceCollection": {
+      "description": "The collection of Media Service resources.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/MediaService"
+          }
+        },
+        "count": {
+          "description": "The number of Media Services in the collection.",
+          "type": "integer",
+          "format": "int32"
+        },
+        "nextLink": {
+          "description": "The continuation token.",
+          "type": "string"
+        }
+      }
+    },
+    "MediaServiceProperties": {
+      "description": "The additional properties of a Media Service resource.",
+      "properties": {
+        "apiEndpoints": {
+          "description": "The Media Services REST API endpoints for this resource.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ApiEndpoint"
+          }
+        },
+        "storageAccounts": {
+          "description": "The storage accounts for this resource.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/StorageAccount"
+          }
+        }
+      }
+    },
+    "RegenerateKeyInput": {
+      "description": "The request body for a RegenerateKey API.",
+      "properties": {
+        "keyType": {
+          "description": "The keyType must be 'Primary' or 'Secondary', indicating which key you want to regenerate.",
+          "type": "string"

```

Comment:
Yes, this should be an enum. Will update.

From: John Hart [mailto:notifications@github.com]
Sent: Wednesday, June 29, 2016 1:16 PM
To: Azure/azure-rest-api-specs <azure-rest-api-specs@noreply.github.com>
Cc: Brian Blum <brianbl@microsoft.com>; Mention <mention@noreply.github.com>
Subject: Re: [Azure/azure-rest-api-specs] Media Services version 2015-10-01 (#376)


In arm-mediaservices/2015-10-01/swagger/media.json<https://github.com/Azure/azure-rest-api-specs/pull/376#discussion_r69018838>:

> +        },

> +        "storageAccounts": {

> +          "description": "The storage accounts for this resource.",

> +          "type": "array",

> +          "items": {

> +            "$ref": "#/definitions/StorageAccount"

> +          }

> +        }

> +      }

> +    },

> +    "RegenerateKeyInput": {

> +      "description": "The request body for a RegenerateKey API.",

> +      "properties": {

> +        "keyType": {

> +          "description": "The keyType must be 'Primary' or 'Secondary', indicating which key you want to regenerate.",

> +          "type": "string"

If Primary and Secondary are the only options should keyType be an enum?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/Azure/azure-rest-api-specs/pull/376/files/5e6e26edeffb270c3e702bbd1756d46a7559d074#r69018838>, or mute the thread<https://github.com/notifications/unsubscribe/AEH9bFyMNOgi4b8veveqDWcKygrGCHusks5qQtKMgaJpZM4JAhvv>.